kibbeh/src/ui/NativeRadio.tsx

```diff
@@ -1,116 +1,90 @@
-
-  const setSetttings = (e: React.MouseEvent) => {
-    const { current } = radioRef;
-    if (current && !current.contains(e.target as any)) {
-      setRadioSelected(false);
-    }
-
-    console.log(radioRef.current?.value);
-  };
-
+  onClick,
+  num,
+}) => {
   return (
-    <React.Fragment>
-      <div
-        style={{
-          width: "601px",
-          height: "64px",
-          borderRadius: "8px",
-          background: "#151A21",
-        }}
-        className="flex justify-between mt-4"
-      >
-        <div
-          style={{ marginTop: "16px", marginBottom: "38px" }}
-          className={icon ? `flex flex-none absolute ml-3 mr-3` : `absolute `}
-        >
-          {icon}
-        </div>
-        <div className={icon ? `ml-6 mt-2 mb-2 relative` : `ml-3 mt-2 mb-2`}>
-          <span
-            className={
-              radioSelected !== checked
-                ? `text-primary-100 font-bold`
-                : `${optionalChangedColor} font-bold`
-            }
+    <button
+      className="w-full flex px-3 py-2 bg-primary-900 rounded-8 justify-between group"
+      onClick={onClick ? () => onClick(num) : undefined}
+    >
+      <div className="flex">
+        {icon ? (
+          <div className="mr-3 mt-1.5">
+            {React.cloneElement(icon, { width: 10, height: 10 })}
+          </div>
+        ) : null}
+        <div className="flex flex-col items-start">
+          <div
+            className={`font-bold group-hover:text-primary-100 transition duration-100 ${
+              checked ? "text-primary-100" : "text-primary-300"
+            }`}
           >
             {title}
-          </span>
-          <p className="text-primary-300">{description}</p>
+          </div>
+          <div className="text-primary-300">{subtitle}</div>
         </div>
-        <div>
-          <label
-            style={{ margin: "22px 15px 22px 0" }}
-            className="flex justify-center content-center"
-          >
-            <input
-              type="radio"
-              ref={radioRef}
-              name={id}
-              className="form-radio text-primary-900"
-              id={id}
-              checked={checked !== radioSelected}
-              onChange={() => setRadioSelected(!radioSelected)}
-              onClick={setSetttings}
-              value={id}
-            />
-          </label>
+      </div>
+      <div className="flex items-center justify-center">
+        <div className="w-4 h-4 relative">
+          <div
+            className={`${
+              checked ? "bg-accent" : ""
+            } w-2 h-2 absolute top-2/4 left-2/4 rounded-full transform -translate-y-1/2 -translate-x-1/2 transition duration-100`}
+          ></div>
+          <div
+            className={`${
+              checked ? "border-accent" : "border-primary-300"
+            } border w-4 h-4 absolute top-2/4 left-2/4 rounded-full transform -translate-y-1/2 -translate-x-1/2 transition duration-100`}
+          ></div>
```

Comment:
transform property to create the radio buttons? This is definitely better than using forms 😆